panic because of unexported field from hpa

This will not heal itself and get into a forever crashloop

```
panic: cannot handle unexported field: {*v2beta1.HorizontalPodAutoscaler}.Spec.Metrics[0].Object.TargetValue.i
consider using AllowUnexported or cmpopts.IgnoreUnexported

goroutine 138 [running]:
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp.invalid.apply(0xc000e7c930, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/options.go:208 +0xf5
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp.(*state).tryOptions(0xc000e7c930, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x13a44c0, 0x11acfc0, 0x0)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/compare.go:307 +0x149
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp.(*state).compareAny(0xc000e7c930, 0x11acfc0, 0xc000b301f0, 0x1b9, 0x11acfc0, 0xc0006161f0, 0x1b9)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/compare.go:202 +0x336
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp.(*state).compareStruct(0xc000e7c930, 0x11a7820, 0xc000b301f0, 0x199, 0x11a7820, 0xc0006161f0, 0x199, 0x13a44c0, 0x11a7820)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/compare.go:503 +0x169
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp.(*state).compareAny(0xc000e7c930, 0x11a7820, 0xc000b301f0, 0x199, 0x11a7820, 0xc0006161f0, 0x199)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/compare.go:272 +0x2522
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp.(*state).compareStruct(0xc000e7c930, 0x1185b20, 0xc000b301b0, 0x199, 0x1185b20, 0xc0006161b0, 0x199, 0x13a44c0, 0x1185b20)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/compare.go:503 +0x169
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp.(*state).compareAny(0xc000e7c930, 0x1185b20, 0xc000b301b0, 0x199, 0x1185b20, 0xc0006161b0, 0x199)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/compare.go:272 +0x2522
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp.(*state).compareAny(0xc000e7c930, 0x1193b20, 0xc000e1a010, 0x196, 0x1193b20, 0xc0004bc580, 0x196)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/compare.go:244 +0x1af5
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp.(*state).compareStruct(0xc000e7c930, 0x1185920, 0xc000e1a000, 0x199, 0x1185920, 0xc0004bc570, 0x199, 0x13a44c0, 0x1185920)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/compare.go:503 +0x169
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp.(*state).compareAny(0xc000e7c930, 0x1185920, 0xc000e1a000, 0x199, 0x1185920, 0xc0004bc570, 0x199)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/compare.go:272 +0x2522
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp.(*state).statelessCompare(0xc000e7c930, 0x1185920, 0xc000e1a000, 0x199, 0x1185920, 0xc0004bc570, 0x199, 0xc00022e128, 0xc000215ec0)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/compare.go:176 +0xc9
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp.(*state).compareArray.func1(0x0, 0x0, 0x0, 0x1e878c0)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/compare.go:398 +0x108
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/internal/diff.Difference(0x1, 0x1, 0xc0002160b0, 0x1, 0x0, 0x0)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/internal/diff/diff.go:212 +0x244
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp.(*state).compareArray(0xc000e7c930, 0x1052b80, 0xc00022e148, 0x197, 0x1052b80, 0xc0001c0bc8, 0x197, 0x13a44c0, 0x1052b80)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/compare.go:396 +0x23d
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp.(*state).compareAny(0xc000e7c930, 0x1052b80, 0xc00022e148, 0x197, 0x1052b80, 0xc0001c0bc8, 0x197)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/compare.go:266 +0xd10
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp.(*state).compareStruct(0xc000e7c930, 0x1171100, 0xc00022e108, 0x199, 0x1171100, 0xc0001c0b88, 0x199, 0x13a44c0, 0x1171100)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/compare.go:503 +0x169
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp.(*state).compareAny(0xc000e7c930, 0x1171100, 0xc00022e108, 0x199, 0x1171100, 0xc0001c0b88, 0x199)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/compare.go:272 +0x2522
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp.(*state).compareStruct(0xc000e7c930, 0x1171020, 0xc00022e000, 0x199, 0x1171020, 0xc0001c0a80, 0x199, 0x13a44c0, 0x1171020)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/compare.go:503 +0x169
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp.(*state).compareAny(0xc000e7c930, 0x1171020, 0xc00022e000, 0x199, 0x1171020, 0xc0001c0a80, 0x199)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/compare.go:272 +0x2522
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp.(*state).compareAny(0xc000e7c930, 0x1208a60, 0xc00022e000, 0x16, 0x1208a60, 0xc0001c0a80, 0x16)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/compare.go:244 +0x1af5
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp.Equal(0x1208a60, 0xc00022e000, 0x1208a60, 0xc0001c0a80, 0xc000b9e240, 0x2, 0x2, 0x0)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/compare.go:86 +0x16b
github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp.Diff(0x1208a60, 0xc00022e000, 0x1208a60, 0xc0001c0a80, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/github.com/google/go-cmp/cmp/compare.go:100 +0x129
github.com/zalando-incubator/stackset-controller/controller.(*stacksReconciler).manageAutoscaling(0xc0002198a0, 0xc0006d9de4, 0x5, 0xc0006d9df0, 0xe, 0xc0003de200, 0x20, 0x0, 0x0, 0xc0006d9e60, ...)
        /go/src/github.com/zalando-incubator/stackset-controller/controller/stack.go:339 +0x25d
github.com/zalando-incubator/stackset-controller/controller.(*stacksReconciler).manageDeployment(0xc0002198a0, 0xc0006d9de4, 0x5, 0xc0006d9df0, 0xe, 0xc0003de200, 0x20, 0x0, 0x0, 0xc0006d9e60, ...)
        /go/src/github.com/zalando-incubator/stackset-controller/controller/stack.go:213 +0x4dd
github.com/zalando-incubator/stackset-controller/controller.(*stacksReconciler).manageStack(0xc0002198a0, 0xc0006d9de4, 0x5, 0xc0006d9df0, 0xe, 0xc0003de200, 0x20, 0x0, 0x0, 0xc0006d9e60, ...)
        /go/src/github.com/zalando-incubator/stackset-controller/controller/stack.go:54 +0x100
github.com/zalando-incubator/stackset-controller/controller.(*StackSetController).ReconcileStacks(0xc0001ed2c0, 0x1223a9d, 0x8, 0x12284fe, 0xe, 0xc000053700, 0x18, 0x0, 0x0, 0xc0000ed359, ...)
        /go/src/github.com/zalando-incubator/stackset-controller/controller/stack.go:49 +0x2ac
github.com/zalando-incubator/stackset-controller/controller.(*StackSetController).Run.func1(0x8, 0x12a2af0)
        /go/src/github.com/zalando-incubator/stackset-controller/controller/stackset.go:105 +0x19d
github.com/zalando-incubator/stackset-controller/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1(0xc00071f4a0, 0xc00071f7a0)
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/golang.org/x/sync/errgroup/errgroup.go:58 +0x57
created by github.com/zalando-incubator/stackset-controller/vendor/golang.org/x/sync/errgroup.(*Group).Go
        /go/src/github.com/zalando-incubator/stackset-controller/vendor/golang.org/x/sync/errgroup/errgroup.go:55 +0x66
```

--- a/controller/stack.go
+++ b/controller/stack.go
@@ -336,7 +336,14 @@ func (c *stacksReconciler) manageAutoscaling(sc StackContainer, deployment *apps
 		}
 	} else {
 		if !equality.Semantic.DeepEqual(origHPA, hpa) {
-			c.logger.Debugf("HPA %s/%s changed: %s", hpa.Namespace, hpa.Name, cmp.Diff(origHPA, hpa))
+			c.logger.Debugf("HPA %s/%s changed: %s",
+				hpa.Namespace, hpa.Name,
+				cmp.Diff(
+					origHPA,
+					hpa,
+					cmpopts.IgnoreUnexported(resource.Quantity{}),
+				),
+			)
 			c.recorder.Eventf(&stack,
 				apiv1.EventTypeNormal,
 				"UpdateHPA",

